src/Core/Grid/Query/Builder/CustomerQueryBuilder.php

```diff
@@ -0,0 +1,201 @@
+     */
+    private $contextLangId;
+
+    /**
+     * @var int[]
+     */
+    private $contextShopIds;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param int $contextShopId
+     * @param int $contextLangId
+     * @param int[] $contextShopIds
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        $contextShopId,
+        $contextLangId,
+        $contextShopIds
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->contextShopId = $contextShopId;
+        $this->contextLangId = $contextLangId;
+        $this->contextShopIds = $contextShopIds;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $searchQueryBuilder = $this->getCustomerQueryBuilder($searchCriteria)
+            ->select('c.id_customer, c.firstname, c.lastname, c.email, c.active, c.newsletter, c.optin')
+            ->addSelect('c.date_add, gl.name as social_title, s.name as shop_name, c.company')
+        ;
+
+        $this->appendTotalSpentQuery($searchQueryBuilder);
+        $this->appendLastVisitQuery($searchQueryBuilder);
+
+        return $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $countQueryBuilder = $this->getCustomerQueryBuilder($searchCriteria)
+            ->select('COUNT(*)')
+        ;
+
+        return $countQueryBuilder;
+    }
+
+    private function getCustomerQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'customer', 'c')
+            ->leftJoin(
+                'c',
+                $this->dbPrefix . 'gender_lang',
+                'gl',
+                'c.id_gender = gl.id_gender AND gl.id_lang = :context_lang_id'
+            )
+            ->leftJoin(
+                'c',
+                $this->dbPrefix . 'shop',
+                's',
+                'c.id_shop = s.id_shop'
+            )
+            ->where('c.deleted = 0')
+            ->andWhere('c.id_shop IN (:context_shop_ids)')
+            ->setParameter('context_shop_ids', $this->contextShopIds, Connection::PARAM_INT_ARRAY)
+            ->setParameter('context_lang_id', $this->contextLangId)
+        ;
+
+        $this->applyFilters($searchCriteria->getFilters(), $queryBuilder);
+
+        return $queryBuilder;
+    }
+
+    private function appendTotalSpentQuery(QueryBuilder $queryBuilder)
+    {
+        $totalSpentQueryBuilder = $this->connection->createQueryBuilder()
+            ->select('SUM(total_paid_real / conversion_rate)')
+            ->from($this->dbPrefix . 'orders', 'o')
+            ->where('o.id_customer = c.id_customer')
+            ->andWhere('o.id_shop IN (:context_shop_ids)')
+            ->andWhere('o.valid = 1')
```

Comment:
`o.valid is TRUE` => `'valid' => array('type' => self::TYPE_BOOL),`

Reply:
i dont really get your comments. :/

Reply:
it's a boolean not an integer. If we plan to move to another database like postgresql, nothing will work :P

Reply:
> we plan to move to another database like postgresql

we better dont. :D

i could do it with parameter binding, but is it worth?

Reply:
When I said `move` i wanted to say `allow` my bad :P 
But if we have a boolean value we need to be consistent and use boolean in our requests.